api/nakadi-event-bus-api.yaml

```diff
@@ -0,0 +1,486 @@
+        - name: stream_limit
+          in: query
+          description: |
+            Maximum number of `Event`s in this stream. If 0 or undefined, will stream indefinately.
+            Stream initialization will fail if `stream_limit` is lower than `batch_limit`.
+          type: integer
+          format: int32
+          required: false
+        - name: batch_flush_timeout
+          in: query
+          description: |
+            Maximum time in seconds to wait for the flushing of each chunk; if the `batch_limit` is reached before this
+            time is reached the messages are immediately flushed to the client.
+            A value of 0 will flush every batch events immediatly as soon as they come in.
+          type: integer
+          format: int32
+          required: false
+        - name: stream_timeout
+          in: query
+          description: |
+            Maximum time in seconds a stream will live before being interrupted. If unspecified will stream
+            indefinately.
+            If this timeout is reached any pending messages (in the sense of `stream_limit`) will be flushed to the
+            client.
+            Stream initialization will fail if `stream_timeout` is lower than `batch_flush_timeout`.
+          type: integer
+          format: int32
+          required: false
+        - name: batch_keep_alive_limit
+          in: query
+          description: |
+            Maximum number of keep-alive messages to get in a row
+            before closing the connection.
+            Unlimited by default.
+          type: integer
+          format: int32
+          required: false
+        - name: Accept-Encoding
+          in: header
+          required: false
+          description: '"If set the client requests the body to be gzip compressed. For Example "Content-Encoding: gzip"'
+          type: string
+      responses:
+        '200':
+          description: |
+            Starts streaming to the client.  
+            Stream format is a continuous series of
+            `SimpleStreamEvent`s separated by `\n`
+          schema:
+            $ref: '#/definitions/SimpleStreamEvent'
+        '400':
+          description: Bad syntax
+          schema:
+            $ref: '#/definitions/Problem'
+        '401':
+          description: Not authenticated
+          schema:
+            $ref: '#/definitions/Problem'
+        '404':
+          description: Not found. Typically in case of requesting from a non existing topic of partition.
+          schema:
+            $ref: '#/definitions/Problem'
+        '500':
+          description: Not found. Typically in case of requesting from a non existing topic of partition.
+          schema:
+            $ref: '#/definitions/Problem'
+
+
+# ################################### #
+#                                     #
+#             Definitions             #
+#                                     #
+# ################################### #
+
+definitions:
+  Event:
```

Comment:
+1 -- some are definitely required (like 'event_type') others are provided by Nakadi (like 'created').
Since you use the same structure for POST and GET required attributes are requested as input params provided by producer, whereas non-required and read-only attributes are computed by Nakadi.

This crucial structure definitely needs clearer description of the different attributes (semantics and pragmatics) and all attributes should also armed with good examples.